src/library-authoring/library-access/UserAccessContainer.jsx

```diff
@@ -0,0 +1,185 @@
+            </span>
+          </Col>
+          {isAdmin && (
+          <Col xs={12} md={6}>
+            <Row>
+              {(user.access_level === LIBRARY_ACCESS.ADMIN) && adminLocked && (
+              <Col xs={12} className="text-center text-md-right">
+                <small>{intl.formatMessage(messages['library.access.info.admin_unlock'])}</small>
+              </Col>
+              )}
+              {user.access_level === LIBRARY_ACCESS.ADMIN && multipleAdmins && (
+              <Col xs={10} className="text-left text-md-right">
+                <Button size="lg" variant="secondary" onClick={() => setShowDeprivModal(true)}>
+                  {intl.formatMessage(messages['library.access.user.remove_admin'])}
+                </Button>
+                <Modal
+                  open={showDeprivModal}
+                  title={intl.formatMessage(messages['library.access.modal.remove_admin.title'])}
+                  onClose={() => setShowDeprivModal(false)}
+                  body={(
+                    <div>
+                      <p>
+                        {intl.formatMessage(
+                          messages['library.access.modal.remove_admin.body'],
+                          { library: library.title, email: user.email },
+                        )}
+                      </p>
+                    </div>
+                  )}
+                  buttons={[
+                    <Button
+                      onClick={() => setAccessLevel(LIBRARY_ACCESS.STAFF).then(setShowDeprivModal(false))}
+                    >
+                      {intl.formatMessage(commonMessages['library.common.forms.button.yes'])}
+                    </Button>,
+                  ]}
+                />
+              </Col>
+              )}
+              {user.access_level === LIBRARY_ACCESS.USER && (
+              <Col xs={10} className="text-left text-md-right">
+                <Button size="lg" variant="primary" onClick={() => setAccessLevel(LIBRARY_ACCESS.STAFF)}>
+                  {intl.formatMessage(messages['library.access.user.add_author'])}
+                </Button>
+              </Col>
+              )}
+              {user.access_level === LIBRARY_ACCESS.STAFF && (
+                <>
+                  <Col xs={5} className="text-left text-md-right pl-md-1">
+                    <Button size="lg" variant="secondary" onClick={() => setAccessLevel(LIBRARY_ACCESS.USER)}>
+                      {intl.formatMessage(messages['library.access.user.remove_author'])}
+                    </Button>
+                  </Col>
+                  <Col xs={5} className="text-left text-md-right pl-md-1">
+                    <Button size="lg" variant="primary" onClick={() => setAccessLevel(LIBRARY_ACCESS.ADMIN)}>
+                      {intl.formatMessage(messages['library.access.user.add_admin'])}
+                    </Button>
+                  </Col>
+                </>
+              )}
```

Comment:
Why is a Staff user able to elevate themselves to Admin level?

I was able to do confirm that it's possible locally, so apparently the API is fine with it as well. Is this existing behavior we brought over from V1 libraries :open_mouth: ?

Reply:
*global* staff users can administer any course or library on the system; is that what you mean? https://github.com/edx/edx-platform/blob/8e1ea3e53637485cc016b1d9fd569263277ba362/openedx/core/djangoapps/content_libraries/permissions.py#L89-L91

People with "Author" or "Read-only" permission in this specific library definitely shouldn't be able to elevate their own permission.

Reply:
@bradenmacdonald is right here. I went back and tried granting staff access to honor@ and when logged in as them, I could not elevate my own privileges. I could only do so as a global staff user, which is how it should work.

Reply:
Got it, thank you for clarifying that @Kelketek .